src/proxy/sessions.rs

```diff
@@ -127,14 +127,33 @@ impl Session {
             metrics: args.metrics,
             config: args.config.clone(),
             upstream_socket,
-            source: args.source,
+            source: args.source.clone(),
             dest: args.dest,
             created_at: Instant::now(),
             expiration,
             shutdown_tx,
         };
         tracing::debug!(source = %s.source, dest = ?s.dest, "Session created");
 
+        if let Some(mmdb) = &*crate::MaxmindDb::instance() {
```

Comment:
Looks good - just so I'm super clear, we're only tracing this information on Session creation? (Which means this can't be a filter either, which is fair enough. i did wonder at one point if #586 should maybe have a "Session creation Filter/Plugin" - which this could have leaned into, but not sure if that's a good idea).

Do you think we'll want to also attach this information to any other traces that may come through this particular session?

Only other thought: We need some docs please 😃

Reply:
Yeah for now I don't see much point in having it as a filter, as it's something that you want to only happen to successful connections, which is easy to guarentee here. We can always change it later.

> Do you think we'll want to also attach this information to any other traces that may come through this particular session?

I think we can if we find it useful. Right now I'm just starting with Quilkin outputting this info and then trying to use it, and seeing what it needs from there.

> Only other thought: We need some docs please 😃

Added